code/modules/mob/living/blood.dm

```diff
@@ -36,50 +37,56 @@
 		if(blood_volume < BLOOD_VOLUME_NORMAL)
 			blood_volume += 0.1 // regenerate blood VERY slowly
 
-		var/b_volume = blood_volume
-
 		// Damaged heart virtually reduces the blood volume, as the blood isn't
 		// being pumped properly anymore.
 		if(species && species.has_organ["heart"])
 			var/datum/internal_organ/heart/heart = internal_organs_by_name["heart"]
 
 			if(!heart)
-				b_volume = 0
-			else if(reagents.get_reagent_amount("peridaxon") >= 0.05)
-				b_volume *= 1
-			else if(heart.damage > 1 && heart.damage < heart.min_bruised_damage)
-				b_volume *= 0.8
-			else if(heart.damage >= heart.min_bruised_damage && heart.damage < heart.min_broken_damage)
-				b_volume *= 0.6
-			else if(heart.damage >= heart.min_broken_damage && heart.damage < INFINITY)
-				b_volume *= 0.3
+				heart_multi += 1 //you'd die in seconds but you can't remove internal organs even with varediting.
+
+			if((!reagents.get_reagent_amount("peridaxon") >= 0.05) && heart.damage > 1)
+				if(heart.damage < heart.min_bruised_damage)
+					heart_multi = 0.9
+					blood_volume -= 0.1 //nulls regeneration
+				else if(heart.damage >= heart.min_bruised_damage && heart.damage < heart.min_broken_damage)
+					heart_multi = 0.7
+					blood_volume -= 0.5
+				else if(heart.damage >= heart.min_broken_damage && heart.damage < INFINITY)
+					heart_multi = 0.5
+					blood_volume -= 1.3
```

Comment:
Couldn't this end up in a negative blood amount?

Reply:
Handle blood doesn't process when you are dead on in cryo.

Reply:
I don't think this is a big issue, but it's still good practice. Everywhere else that is uncommented in the code does something similar to `blood_volume = max(0, blood_volume - amount)`. Might want to implement something like that.